In [3]:
%matplotlib inline
import pandas as pd
import os
import seaborn as sns
import numpy as np
import matplotlib.pyplot as plt

In [ ]:
# from IPython.core.display import HTML
# css = open('style-table.css').read() + open('style-notebook.css').read()
# HTML('<style>{}</style>'.format(css))

In [12]:
titles = pd.read_csv('/titles.csv')
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

In [13]:

titles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 244914 entries, 0 to 244913
Data columns (total 2 columns):
 #   Column  Non-Null Count   Dtype 
---  ------  --------------   ----- 
 0   title   244914 non-null  object
 1   year    244914 non-null  int64 
dtypes: int64(1), object(1)
memory usage: 3.7+ MB


In [14]:
titles.describe()

year
count  244914.000000
mean     1986.616012
std        29.459912
min      1894.000000
25%      1967.000000
50%      1997.000000
75%      2012.000000
max      2115.000000

In [15]:
titles.duplicated()

0         False
1         False
2         False
3         False
4         False
          ...  
244909    False
244910    False
244911    False
244912    False
244913    False
Length: 244914, dtype: bool

### How many movies are listed in the titles dataframe?

In [16]:
titles['title'].unique().shape

(223828,)

### There are 223828 unique movies in 244914 movie list in the title

### What are the earliest two films listed in the titles dataframe?

In [17]:
titles.sort_values(by =['year'],ascending= True).head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

####  __Miss Jerry__ and __The Startled Lover__ are the two earliest movies 

### How many movies have the title "Hamlet"?

In [18]:
a =titles[titles.title.str.contains('Hamlet', case=False)]

In [19]:
a.sort_values(by =['year'],ascending= True)

title  year
52521                                    Hamlet  1910
198234                                   Hamlet  1911
39916                                    Hamlet  1913
214820                       A Sagebrush Hamlet  1919
10637                                    Hamlet  1921
48703                        Han, hun og Hamlet  1922
94884                        Han, hun og Hamlet  1932
101728                                   Hamlet  1948
172640                                   Hamlet  1954
176125                              Hamlet (II)  1964
161324                                   Hamlet  1964
67406                                    Hamlet  1969
122827                                   Hamlet  1973
148503      Predstava 'Hamleta' u Mrdusi Donjoj  1974
239719              Intikam Melegi/Kadin Hamlet  1976
133921                                   Hamlet  1976
82130                                    Hamlet  1987
52097                    Hamlet liikemaailmassa  1987
104397                Hamlet: Prince of Denmark  1989
192142                                   Hamlet  1990
223833                             H for Hamlet  1993
48886                     Green Eggs and Hamlet  1995
200307                              Fuck Hamlet  1996
178222                                   Hamlet  1996
51159                 Hamlet, Prince of Denmark  1997
62000                                    Hamlet  2000
236188                   Hamlet in the Hamptons  2002
84840                                  Hamlet_X  2003
198793                                 Hamlet X  2004
109416          Dogg's Hamlet, Cahoot's Macbeth  2005
94411                               Hamlet (II)  2005
98431   The Tragedy of Hamlet Prince of Denmark  2007
199932                             Hamlet (III)  2007
244624                              Hamlet (II)  2007
190799                Hamlet the Vampire Slayer  2008
177050                                 Hamlet 2  2008
189093                                   Hamlet  2009
156655                           Kitchen Hamlet  2009
194338                     Hamlet Demonstration  2011
173033                                   Hamlet  2011
46189                             Zombie Hamlet  2012
102779                           Hamlet Unbound  2012
128792                      Harry, Hamlet and I  2013
178829                            Hamlet A.D.D.  2014
202025                                   Hamlet  2014
154029                                   Hamlet  2015
178529                              Hamlet (II)  2015
138162                 Hamlet, Son of a Kingpin  2015
105669                           Hamlet's Ghost  2015
84606             National Theatre Live: Hamlet  2015
89209         Royal Shakespeare Company: Hamlet  2016
88065                                    Hamlet  2016
6422                                     Hamlet  2017
240540                           Hamlet/Horatio  2017
182847                Hamlet in the Golden Vale  2017
135751                              Hamlet (II)  2017
50258             Hamlet: The Fall of a Sparrow  2018

In [20]:
a.shape

(57, 2)

There are 57 movies in which Hamlet is present.

### How many movies are titled "North by Northwest"?

In [21]:
a =titles[titles.title.str.contains('North by Northwest', case=False)]


In [22]:
a.sort_values(by =['year'],ascending= True)

title  year
74398  North by Northwest  1959

There is only one movie which is released in 1959

### When was the first movie titled "Hamlet" made?

In [23]:
a =titles[titles.title.str.contains('Hamlet', case=False)]

In [24]:
a.sort_values(by =['year'],ascending= True).head(3)

title  year
52521   Hamlet  1910
198234  Hamlet  1911
39916   Hamlet  1913

So the first Hamlet movie is released inn 1910


### List all of the "Treasure Island" movies from earliest to most recent.

In [25]:
a =titles[titles.title.str.contains('Treasure Island', case=False)]

In [26]:
a.sort_values(by =['year'],ascending= True)

title  year
55686                                     Treasure Island  1918
196514                                    Treasure Island  1920
141822                                    Treasure Island  1934
64173                       The Secret of Treasure Island  1938
79701                     Charlie Chan at Treasure Island  1939
77354                                     Treasure Island  1950
34341                           Return to Treasure Island  1954
34288                           Five on a Treasure Island  1957
65051                                     Treasure Island  1972
37238                                     Treasure Island  1973
173051                                    Treasure Island  1985
26111                              Muppet Treasure Island  1996
27076                                     Treasure Island  1999
216970                               Treasure Island (II)  1999
209707  Treasure Island Kids: The Battle of Treasure I...  2006
2830                           Pirates of Treasure Island  2006
112591  Treasure Island Kids: The Mystery of Treasure ...  2006
244541  Treasure Island Kids: The Monster of Treasure ...  2006
155317             National Theatre Live: Treasure Island  2015

### How many movies were made in the year 1950?

In [27]:
# titles[titles['year']=='1950']

titles[(titles == 1950).any(axis=1)].count()

title    1113
year     1113
dtype: int64

### 1113 movies were made in 1950

### How many movies were made in the year 1960?

In [28]:
titles[(titles == 1960).any(axis=1)].count()

title    1549
year     1549
dtype: int64

### 1549 movies were made in 1960

### How many movies were made from 1950 through 1959?

In [34]:
titles[(titles['year']>=1950)&(titles['year']<=1959)]

title  year
5                    Tight Spot  1955
8               Bao dao mei ren  1956
45           Trois de la marine  1957
48                    Hong hong  1956
61              Alarma in munti  1955
...                         ...   ...
244803             Touch and Go  1955
244817            White Feather  1955
244818      Carovana di canzoni  1955
244820                 Doctor X  1950
244841  Matros soshyol na bereg  1957

[12934 rows x 2 columns]

Total number of movies made from 1950 to 1959 are: 12934

### In what years has a movie titled "Batman" been released?

In [44]:
a =titles[titles.title.str.contains('Batman', case=False)]

In [46]:
a.sort_values(by =['year'],ascending= True)

title  year
39349                                        Batman  1943
204905                             Batman and Robin  1949
93271                                Batman Dracula  1964
34251                         Alyas Batman at Robin  1965
180265                                 James Batman  1966
134007                            Batman: The Movie  1966
170948                        Batman Fights Dracula  1967
91440                         Fight! Batman, Fight!  1973
240896                                       Batman  1989
183764                        Alyas Batman en Robin  1991
21158                                Batman Returns  1992
92106                  Batman: Mask of the Phantasm  1993
163887                               Batman Forever  1995
153828                               Batman & Robin  1997
223791                                Batman Begins  2005
3609                                 Batman Reboots  2010
78155                     Batman: The Final Victory  2010
49259                                 Batman Untold  2010
161608                              Batman Revealed  2012
139039                             Batman: Crusader  2014
143064                 Batman Beyond: Rising Knight  2014
56896            Batman v Superman: Dawn of Justice  2016
29920                               Batman: Anarchy  2016
192806           B@: It's a Wonderful Life, Batman!  2016
42483                      Batman: The Killing Joke  2016
130474        Batman: Return of the Caped Crusaders  2016
93833                       B@ (Batman Parody Film)  2016
230776                       Batman: Master of Fear  2017
213182                Batman the Dawn of the Knight  2017
26577                         The LEGO Batman Movie  2017
63068              Batman V Superman: League at War  2018
161657                  Batman the Rise of Red Hood  2018
194288  Scooby-Doo & Batman: the Brave and the Bold  2018
108277                                 Batman Ninja  2018
127552                              Batman Unveiled  2018

### How many roles were there in the movie "Inception"?

In [47]:
cast = pd.read_csv("/cast.csv")
cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4  31.0
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

In [48]:
rel=pd.read_csv("/release_dates.csv")
rel.head()

title  year      country        date
0   #73, Shaanthi Nivaasa  2007        India  2007-06-15
1                   #BKKY  2016     Cambodia  2017-10-12
2                 #Beings  2015      Romania  2015-01-29
3               #Captured  2017          USA  2017-09-05
4  #Ewankosau saranghaeyo  2015  Philippines  2015-01-21

In [51]:
a= cast[cast.title.str.contains('Batman', case=False)]

In [57]:
list_of_char= list(a['character'].unique())

In [66]:
print(f'The no of unique characcters : {len(list_of_char)}')

The no of unique characcters : 899


### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [68]:

print("The number of roles in the movie Inception not ranked by n value are:")
print(len(cast[(cast['title']=='Inception')&(cast['n'].isnull()==True)].type))
print("The roles are displayed below:")
cast[(cast['title']=='Inception')&(cast['n'].isnull()==True)]

The number of roles in the movie Inception not ranked by n value are:
27
The roles are displayed below:


title  year  ...                  character   n
98153    Inception  2010  ...              LAX Passenger NaN
197877   Inception  2010  ...           Mombasan Gambler NaN
374661   Inception  2010  ...              LAX Passenger NaN
497161   Inception  2010  ...             Snow Commander NaN
684098   Inception  2010  ...            Italian Gambler NaN
690634   Inception  2010  ...               Stock Broker NaN
707642   Inception  2010  ...                      Extra NaN
836295   Inception  2010  ...             Bridge Sub Con NaN
1006636  Inception  2010  ...                 Pedestrian NaN
1095095  Inception  2010  ...                Hotel Guest NaN
1108755  Inception  2010  ...      Japanese Landing Crew NaN
1120184  Inception  2010  ...      Mombasan Casino Staff NaN
1189066  Inception  2010  ...     Mombasa Casino Cashier NaN
1248176  Inception  2010  ...  Saito's Bodyguard Officer NaN
1296268  Inception  2010  ...              Street Person NaN
1312335  Inception  2010  ...             French Student NaN
1665939  Inception  2010  ...                A Passer-By NaN
1836655  Inception  2010  ...                 Pedestrian NaN
1921209  Inception  2010  ...             Bridge Sub Con NaN
1999736  Inception  2010  ...             Airport Patron NaN
2127670  Inception  2010  ...             Airport Patron NaN
2238386  Inception  2010  ...                    Denison NaN
2796449  Inception  2010  ...           Mombasan Cashier NaN
2852164  Inception  2010  ...                Stockbroker NaN
2950644  Inception  2010  ...              LAX Passenger NaN
3054447  Inception  2010  ...                   Reporter NaN
3149397  Inception  2010  ...             Airport Patron NaN

[27 rows x 6 columns]

### But how many roles in the movie "Inception" did receive an "n" value?

In [70]:

print("The number of roles in the movie Inception with n value are:")
print(len(cast[(cast['title']=='Inception')&(cast['n'].isnull()==False)].type))
print("The roles are displayed below:")
cast[(cast['title']=='Inception')&(cast['n'].isnull()==False)]

The number of roles in the movie Inception with n value are:
51
The roles are displayed below:


title  year  ...                character     n
149008   Inception  2010  ...    Fischer's Jet Captain  44.0
183937   Inception  2010  ...                 Browning   8.0
260128   Inception  2010  ...           Bridge Sub Con  29.0
326745   Inception  2010  ...                    Miles  11.0
333415   Inception  2010  ...         Elderly Bald Man  20.0
433867   Inception  2010  ...           Bridge Sub Con  26.0
444385   Inception  2010  ...              Bar Sub Con  34.0
517227   Inception  2010  ...           Bridge Sub Con  30.0
590576   Inception  2010  ...                     Cobb   1.0
709436   Inception  2010  ...               Cab Driver  23.0
737322   Inception  2010  ...          Penrose Sub Con  32.0
807795   Inception  2010  ...      Immigration Officer  45.0
812135   Inception  2010  ...          James (3 years)  17.0
832083   Inception  2010  ...            Lobby Sub Con  40.0
833376   Inception  2010  ...                 Old Cobb  50.0
859993   Inception  2010  ...                   Arthur   2.0
910757   Inception  2010  ...                     Nash  12.0
940923   Inception  2010  ...                    Eames   4.0
967176   Inception  2010  ...                   Lawyer  21.0
1170785  Inception  2010  ...                 Thin Man  24.0
1309851  Inception  2010  ...                  Tadashi  13.0
1455534  Inception  2010  ...  Japanese Security Guard  18.0
1615709  Inception  2010  ...           Robert Fischer   7.0
1618527  Inception  2010  ...              Bar Sub Con  35.0
1668790  Inception  2010  ...        James (20 months)  15.0
1701902  Inception  2010  ...        Saito's Attendant  19.0
1812091  Inception  2010  ...              Businessman  47.0
1826027  Inception  2010  ...          Maurice Fischer  10.0
1861317  Inception  2010  ...            Lobby Sub Con  39.0
1876301  Inception  2010  ...                    Yusuf   6.0
2049179  Inception  2010  ...              Businessman  46.0
2250605  Inception  2010  ...         Fischer's Driver  49.0
2267383  Inception  2010  ...              Bar Sub Con  36.0
2406531  Inception  2010  ...                    Saito   5.0
2614557  Inception  2010  ...              Bar Sub Con  38.0
2765969  Inception  2010  ...                      Mal   9.0
2776554  Inception  2010  ...          Penrose Sub Con  31.0
2807441  Inception  2010  ...           Bridge Sub Con  27.0
2932657  Inception  2010  ...       Phillipa (3 years)  14.0
2932664  Inception  2010  ...       Phillipa (5 years)  16.0
3151982  Inception  2010  ...           Bridge Sub Con  28.0
3158892  Inception  2010  ...          Penrose Sub Con  33.0
3203564  Inception  2010  ...            Lobby Sub Con  42.0
3225843  Inception  2010  ...            Lobby Sub Con  41.0
3343512  Inception  2010  ...              Bar Sub Con  37.0
3359825  Inception  2010  ...         Flight Attendant  22.0
3387147  Inception  2010  ...                  Ariadne   3.0
3444628  Inception  2010  ...            Lobby Sub Con  43.0
3473041  Inception  2010  ...            Private Nurse  48.0
3480170  Inception  2010  ...                   Blonde  25.0
3731263  Inception  2010  ...                  Old Mal  51.0

[51 rows x 6 columns]

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [71]:
print("The cast of North by Northwest in their correct n-value order value with n values are:")
print(cast[(cast['title']=="North by Northwest")&(cast['n'].isnull()==False)].sort_values(by=['n']).name)
print("The cast is displayed below:")
cast[(cast['title']=="North by Northwest")&(cast['n'].isnull()==False)].sort_values(by=['n'])

The cast of North by Northwest in their correct n-value order value with n values are:
871426               Cary Grant
3518333         Eva Marie Saint
1457165             James Mason
3157232     Jessie Royce Landis
356707           Leo G. Carroll
3048272    Josephine Hutchinson
1695173             Philip Ober
1275337           Martin Landau
2447081           Adam Williams
1811387            Edward Platt
665768        Robert Ellenstein
2295714            Les Tremayne
464736          Philip Coolidge
1508344           Patrick McVey
206471             Edward Binns
1384780               Ken Lynch
Name: name, dtype: object
The cast is displayed below:


title  year  ...                    character     n
871426   North by Northwest  1959  ...           Roger O. Thornhill   1.0
3518333  North by Northwest  1959  ...                  Eve Kendall   2.0
1457165  North by Northwest  1959  ...              Phillip Vandamm   3.0
3157232  North by Northwest  1959  ...              Clara Thornhill   4.0
356707   North by Northwest  1959  ...                The Professor   5.0
3048272  North by Northwest  1959  ...                Mrs. Townsend   6.0
1695173  North by Northwest  1959  ...              Lester Townsend   7.0
1275337  North by Northwest  1959  ...                      Leonard   8.0
2447081  North by Northwest  1959  ...                     Valerian   9.0
1811387  North by Northwest  1959  ...              Victor Larrabee  10.0
665768   North by Northwest  1959  ...                        Licht  11.0
2295714  North by Northwest  1959  ...                   Auctioneer  12.0
464736   North by Northwest  1959  ...                    Dr. Cross  13.0
1508344  North by Northwest  1959  ...               Sergeant Flamm  14.0
206471   North by Northwest  1959  ...               Captain Junket  15.0
1384780  North by Northwest  1959  ...  Charley - Chicago Policeman  16.0

[16 rows x 6 columns]

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [84]:

print("The entire cast of 1972 film Seluth order by n value is:")
print((cast[(cast['title']=="Sleuth")&(cast['year']==1972)]).sort_values(by=['n']).name)
print("The entire cast is displayed below:")
cast[(cast['title']=="Sleuth")&(cast['year']==1972)].sort_values(by=['n'])

The entire cast of 1972 film Seluth order by n value is:
1705394      Laurence Olivier
326773          Michael Caine
373876         Alec Cawthorne
1465483    John (II) Matthews
2724553    Eve (III) Channing
1448690          Teddy Martin
Name: name, dtype: object
The entire cast is displayed below:


title  year  ...                   character    n
1705394  Sleuth  1972  ...                 Andrew Wyke  1.0
326773   Sleuth  1972  ...                 Milo Tindle  2.0
373876   Sleuth  1972  ...           Inspector Doppler  3.0
1465483  Sleuth  1972  ...  Detective Sergeant Tarrant  4.0
2724553  Sleuth  1972  ...             Marguerite Wyke  5.0
1448690  Sleuth  1972  ...      Police Constable Higgs  6.0

[6 rows x 6 columns]

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [83]:

print("The entire cast of 2007 film Seluth order by n value is:")
print(cast[(cast['title']=="Sleuth")&(cast['year']==2007)].sort_values(by=['n']).name)
print("The entire cast is displayed below:")
cast[(cast['title']=="Sleuth")&(cast['year']==2007)].sort_values(by=['n'])

The entire cast of 2007 film Seluth order by n value is:
326774           Michael Caine
1293634               Jude Law
1805907          Harold Pinter
260242         Kenneth Branagh
373877     Alec (II) Cawthorne
2724552      Eve (II) Channing
3370424      Carmel O'Sullivan
Name: name, dtype: object
The entire cast is displayed below:


title  year                 name     type          character    n
326774   Sleuth  2007        Michael Caine    actor             Andrew  1.0
1293634  Sleuth  2007             Jude Law    actor               Milo  2.0
1805907  Sleuth  2007        Harold Pinter    actor        Man on T.V.  3.0
260242   Sleuth  2007      Kenneth Branagh    actor  Other Man on T.V.  NaN
373877   Sleuth  2007  Alec (II) Cawthorne    actor  Inspector Doppler  NaN
2724552  Sleuth  2007    Eve (II) Channing  actress    Marguerite Wyke  NaN
3370424  Sleuth  2007    Carmel O'Sullivan  actress             Maggie  NaN

### How many roles were credited in the silent 1921 version of Hamlet?

In [82]:

print("The number of roles that were credited in the silent 1921 version of Hamlet are:")
print(len(cast[(cast['title']=='Hamlet')&(cast['year']==1921)]))
print("The roles are displayed below:")
cast[(cast['title']=='Hamlet')&(cast['year']==1921)]

The number of roles that were credited in the silent 1921 version of Hamlet are:
9
The roles are displayed below:


title  year                    name     type         character    n
7789     Hamlet  1921        Fritz Achterberg    actor       Fortinbras,  9.0
458904   Hamlet  1921            Paul Conradi    actor      K?nig Hamlet  2.0
555034   Hamlet  1921        Anton De Verdier    actor           Laertes  7.0
1135279  Hamlet  1921         Hans Junkermann    actor          Polonius  6.0
2185558  Hamlet  1921            Heinz Stieda    actor           Horatio  5.0
2375993  Hamlet  1921  Eduard von Winterstein    actor          Claudius  4.0
2664423  Hamlet  1921         Mathilde Brandt  actress  K?nigin Gertrude  3.0
3062023  Hamlet  1921          Lilly Jacobson  actress           Ophelia  8.0
3355176  Hamlet  1921            Asta Nielsen  actress            Hamlet  1.0

### How many roles were credited in Branagh’s 1996 Hamlet?

In [81]:
print("The number of roles that were credited in the Branagh's 1996 Hamlet are:")
print(len(cast[(cast['title']=='Hamlet')&(cast['year']==1996)]))
print("The roles are displayed below:")
cast[(cast['title']=='Hamlet')&(cast['year']==1996)]

The number of roles that were credited in the Branagh's 1996 Hamlet are:
54
The roles are displayed below:


title  year  ...                 character     n
1731     Hamlet  1996  ...     Attendant to Claudius   1.0
95861    Hamlet  1996  ...        English Ambassador   2.0
158642   Hamlet  1996  ...        Second Gravedigger  37.0
214473   Hamlet  1996  ...     Attendant to Claudius   3.0
219554   Hamlet  1996  ...  Ghost of Hamlet's Father   4.0
250773   Hamlet  1996  ...             Cadet in Play   NaN
260225   Hamlet  1996  ...                    Hamlet   5.0
272749   Hamlet  1996  ...                  Polonius   6.0
294652   Hamlet  1996  ...                    Priest   7.0
319501   Hamlet  1996  ...     Attendant to Claudius   8.0
344365   Hamlet  1996  ...               Laertes Mob   NaN
500916   Hamlet  1996  ...         First Gravedigger  10.0
518572   Hamlet  1996  ...             Stage Manager  11.0
576147   Hamlet  1996  ...                  Reynaldo  13.0
598618   Hamlet  1996  ...              Guildenstern  14.0
604339   Hamlet  1996  ...                    Yorick  15.0
656817   Hamlet  1996  ...                  Lucianus  17.0
669232   Hamlet  1996  ...            Old Fortinbras   NaN
704270   Hamlet  1996  ...                   Horatio  18.0
708468   Hamlet  1996  ...                 Francisco  19.0
827066   Hamlet  1996  ...                     Priam  21.0
990819   Hamlet  1996  ...               Player King  23.0
1074691  Hamlet  1996  ...                 Cornelius  24.0
1082161  Hamlet  1996  ...                  Claudius  25.0
1210509  Hamlet  1996  ...      Fortinbras's Captain  27.0
1316617  Hamlet  1996  ...                 Marcellus  29.0
1400732  Hamlet  1996  ...         Elsinore Courtier   NaN
1417360  Hamlet  1996  ...                   Laertes  31.0
1489191  Hamlet  1996  ...                  Barnardo  30.0
1549128  Hamlet  1996  ...                Old Norway  33.0
1556082  Hamlet  1996  ...                Sailor Two  34.0
1588218  Hamlet  1996  ...                   Pyrrhus   NaN
2040040  Hamlet  1996  ...                Young Lord  38.0
2052796  Hamlet  1996  ...                   Boatman   NaN
2067657  Hamlet  1996  ...                Fortinbras  39.0
2151526  Hamlet  1996  ...               Rosencrantz  40.0
2154960  Hamlet  1996  ...      Fortinbras's Captain  32.0
2221910  Hamlet  1996  ...              Young Hamlet  41.0
2259675  Hamlet  1996  ...              First Player  42.0
2404091  Hamlet  1996  ...                 Voltimand  43.0
2453099  Hamlet  1996  ...                     Osric  45.0
2500066  Hamlet  1996  ...                Sailor One  47.0
2508377  Hamlet  1996  ...                 Alexander   NaN
2737885  Hamlet  1996  ...                  Gertrude   9.0
2814220  Hamlet  1996  ...                    Hecuba  12.0
2835947  Hamlet  1996  ...     Attendant to Gertrude  16.0
2940033  Hamlet  1996  ...                    Doctor  20.0
2999610  Hamlet  1996  ...              Player Queen  22.0
3119901  Hamlet  1996  ...     Attendant to Gertrude  26.0
3154270  Hamlet  1996  ...     Attendant to Gertrude  28.0
3451003  Hamlet  1996  ...                  Prologue  35.0
3456114  Hamlet  1996  ...                Prostitute   NaN
3728751  Hamlet  1996  ...             Second Player  44.0
3751232  Hamlet  1996  ...                   Ophelia  46.0

[54 rows x 6 columns]

### How many "Hamlet" roles have been listed in all film credits through history?

In [80]:
print("The number of Hamlet roles that have been listed in all film credits throughout the history are:")
print(len(cast[(cast['title']=='Hamlet')]))
print("The roles are displayed below:")
cast[(cast['title']=='Hamlet')]

The number of Hamlet roles that have been listed in all film credits throughout the history are:
342
The roles are displayed below:


title  year              name     type              character     n
1731     Hamlet  1996        Riz Abbasi    actor  Attendant to Claudius   1.0
7789     Hamlet  1921  Fritz Achterberg    actor            Fortinbras,   9.0
11728    Hamlet  2009      Hayden Adams    actor                Laertes   7.0
11729    Hamlet  2009      Hayden Adams    actor                 Player   7.0
14493    Hamlet  1913       Eric Adeney    actor               Reynaldo  14.0
...         ...   ...               ...      ...                    ...   ...
3673234  Hamlet  1969    Jennifer Tudor  actress             Court lady  23.0
3697397  Hamlet  2000      Diane Venora  actress               Gertrude   3.0
3728751  Hamlet  1996     Perdita Weeks  actress          Second Player  44.0
3733637  Hamlet  2015   Katie (II) West  actress                Ophelia   7.0
3751232  Hamlet  1996      Kate Winslet  actress                Ophelia  46.0

[342 rows x 6 columns]

### How many people have played an "Ophelia"?

In [86]:
print("The number of people who played Ophelia are:" , len(cast[cast['character']=='Ophelia']))
# print(len(cast[cast['character']=='Ophelia']))
print("The names are listed below:")
cast[cast['character']=='Ophelia'].name

The number of people who played Ophelia are: 117
The names are listed below:


248251              Amy Bourque
410247           Josie Chisholm
470859          Crystal Cordero
866334            Lucas Grabeel
1364457     Catherine (IV) Lord
                   ...         
3722298         Marsha Warfield
3723039    Jennifer Lynn Warren
3733637         Katie (II) West
3748118       Helen (IV) Wilson
3751232            Kate Winslet
Name: name, Length: 117, dtype: object

### How many people have played a role called "The Dude"?

In [89]:
print("The number of people who played The Dude are:")
print(len(cast[cast['character']=='The Dude']))
print("The names are listed below:")
cast[cast['character']=='The Dude'].name

The number of people who played The Dude are:
19
The names are listed below:


272161             Jeff Bridges
339389              Jordan Cann
498660            Scott Crowell
570128           Chris DeMarcus
610791              Mike Donlin
629473           Devin The Dude
855976       Michael D. Goodwin
1089444     Matthew (XIX) James
1138933      Christopher Kadish
1367230        Jason Loughridge
1386561             Steve Lyons
1512572          Mark Alan Meek
1540933             Jerry Miley
1784519        House Peters Jr.
1824744    Thomas (VIII) Porter
1838960         Jason Priestley
2202150            Bobby Styles
2336261         Dick Van Patten
2419271              Joel Weiss
Name: name, dtype: object

### How many people have played a role called "The Stranger"?

In [90]:
print("The number of people who played The Stranger are:")
print(len(cast[cast['character']=='The Stranger']))
print("The names are listed below:")
cast[cast['character']=='The Stranger'].name

The number of people who played The Stranger are:
212
The names are listed below:


25954      Hassan Mutlag Al-Maraiyeh
35748           Vincent H. Alexander
46418                   Terral Altom
67421                   Tony Anthony
67424                   Tony Anthony
                     ...            
3219011                  Ethel Lynne
3405700               Rosa Maria Paz
3427715             Angela Pleasence
3567516              Dorinea Shirley
3746949    Kerry (II) Willison-Parry
Name: name, Length: 212, dtype: object

### How many roles has Sidney Poitier played throughout his career?

In [91]:
print("The number of roles played by Sidney Poitier are:")
print(len(cast[cast['name']=='Sidney Poitier']))
print("The roles are displayed below:")
cast[cast['name']=='Sidney Poitier']

The number of roles played by Sidney Poitier are:
43
The roles are displayed below:


title  ...     n
1816209               A Patch of Blue  ...   1.0
1816210         A Piece of the Action  ...   1.0
1816211           A Raisin in the Sun  ...   1.0
1816212               A Warm December  ...   1.0
1816213             All the Young Men  ...   2.0
1816214                Band of Angels  ...   3.0
1816215             Blackboard Jungle  ...  10.0
1816216                  Brother John  ...   1.0
1816217         Buck and the Preacher  ...   1.0
1816218      Cry, the Beloved Country  ...   3.0
1816219                Duel at Diablo  ...   2.0
1816220              Edge of the City  ...   2.0
1816221               For Love of Ivy  ...   1.0
1816222                     Go Man Go  ...   3.0
1816223             Good-bye, My Lady  ...   4.0
1816224  Guess Who's Coming to Dinner  ...   2.0
1816225      In the Heat of the Night  ...   1.0
1816226             Let's Do It Again  ...   1.0
1816227           Lilies of the Field  ...   1.0
1816228                 Little Nikita  ...   1.0
1816229                    No Way Out  ...   4.0
1816230                   Paris Blues  ...   3.0
1816231                Porgy and Bess  ...   1.0
1816232                Pressure Point  ...   1.0
1816233              Red Ball Express  ...   5.0
1816234              Sepia Cinderella  ...   NaN
1816235                 Shoot to Kill  ...   1.0
1816236                      Sneakers  ...   4.0
1816237            Something of Value  ...  13.0
1816238          The Bedford Incident  ...   2.0
1816239              The Defiant Ones  ...   2.0
1816240  The Greatest Story Ever Told  ...  21.0
1816241                    The Jackal  ...   3.0
1816242                The Long Ships  ...   2.0
1816243                  The Lost Man  ...   1.0
1816244          The Mark of the Hawk  ...   2.0
1816245              The Organization  ...   1.0
1816246            The Slender Thread  ...   1.0
1816247          The Wilby Conspiracy  ...   1.0
1816248    They Call Me Mister Tibbs!  ...   1.0
1816249             To Sir, with Love  ...   1.0
1816250         Uptown Saturday Night  ...   1.0
1816251                 Virgin Island  ...   3.0

[43 rows x 6 columns]

### How many roles has Judi Dench played?

In [92]:
print("The number of roles played by Judi Dench are:")
print(len(cast[cast['name']=='Judi Dench']))
print("The roles are displayed below:")
cast[cast['name']=='Judi Dench']

The number of roles played by Judi Dench are:
55
The roles are displayed below:


title  ...     n
2814206                        84 Charing Cross Road  ...   3.0
2814207                            A Handful of Dust  ...  13.0
2814208                    A Midsummer Night's Dream  ...  16.0
2814209                           A Room with a View  ...   7.0
2814210                            A Study in Terror  ...   8.0
2814211       Angelina Ballerina: Angelina Sets Sail  ...   2.0
2814212      Branagh Theatre Live: The Winter's Tale  ...   NaN
2814213                                Casino Royale  ...   4.0
2814214                                     Chocolat  ...  18.0
2814215                                    Dead Cert  ...   2.0
2814216                              Die Another Day  ...   6.0
2814217                                       Doogal  ...   7.0
2814218                          Four in the Morning  ...   2.0
2814219                                    GoldenEye  ...   6.0
2814220                                       Hamlet  ...  12.0
2814221                         He Who Rides a Tiger  ...   2.0
2814222                                      Henry V  ...  26.0
2814223                            Home on the Range  ...   7.0
2814224                                         Iris  ...   3.0
2814225                                     J. Edgar  ...  13.0
2814226                                 Jack & Sarah  ...   3.0
2814227                                    Jane Eyre  ...  18.0
2814228                           Ladies in Lavender  ...   1.0
2814229                                       Luther  ...   7.0
2814230  Miss Peregrine's Home for Peculiar Children  ...   4.0
2814231                                    Mrs Brown  ...   1.0
2814232                       Mrs Henderson Presents  ...   1.0
2814233                 Murder on the Orient Express  ...  39.0
2814234                         My Week with Marilyn  ...  23.0
2814235                                         Nine  ...   6.0
2814236                           Notes on a Scandal  ...   1.0
2814237                                    Philomena  ...   1.0
2814238  Pirates of the Caribbean: On Stranger Tides  ...  16.0
2814239                            Pride & Prejudice  ...  20.0
2814240                            Quantum of Solace  ...   4.0
2814241                                         Rage  ...   NaN
2814242                                     Red Joan  ...   NaN
2814243                            Run for Your Wife  ...  40.0
2814244                          Shakespeare in Love  ...  12.0
2814245                                      Skyfall  ...   2.0
2814246                                      Spectre  ...   NaN
2814247                              Stars in Shorts  ...  29.0
2814248                           Tea with Mussolini  ...   2.0
2814249                     The Angelic Conversation  ...   3.0
2814250               The Best Exotic Marigold Hotel  ...   1.0
2814251                    The Chronicles of Riddick  ...   4.0
2814252              The Importance of Being Earnest  ...   5.0
2814253        The Second Best Exotic Marigold Hotel  ...   5.0
2814254                            The Shipping News  ...   3.0
2814255                             The Third Secret  ...  12.0
2814256                      The World Is Not Enough  ...   6.0
2814257                          Tomorrow Never Dies  ...   9.0
2814258                                  Tulip Fever  ...  11.0
2814259                             Victoria & Abdul  ...   1.0
2814260                                     Wetherby  ...   3.0

[55 rows x 6 columns]

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [93]:
print("The list of supporting roles played by Cary Grant in 1940's ordered by year are:")
cast[(cast['name']=='Cary Grant')&(cast['n']==2)&(cast['year']>=1940)&(cast['year']<=1949)].sort_values(by=['year'])

The list of supporting roles played by Cary Grant in 1940's ordered by year are:


title  year        name   type    character    n
871423  My Favorite Wife  1940  Cary Grant  actor   Nick Arden  2.0
871433    Penny Serenade  1941  Cary Grant  actor  Roger Adams  2.0

### List the leading roles that Cary Grant played in the 1940s in order by year.

In [94]:
print(" List the leading roles that Cary Grant played in the 1940s ordered by year are:")
cast[(cast['name']=='Cary Grant')&(cast['n']==1)&(cast['year']>=1940)&(cast['year']<=1949)].sort_values(by=['year'])

 List the leading roles that Cary Grant played in the 1940s ordered by year are:


title  ...    n
871448               The Howards of Virginia  ...  1.0
871407                       His Girl Friday  ...  1.0
871450                The Philadelphia Story  ...  1.0
871438                             Suspicion  ...  1.0
871452                  The Talk of the Town  ...  1.0
871429                 Once Upon a Honeymoon  ...  1.0
871398                     Destination Tokyo  ...  1.0
871422                             Mr. Lucky  ...  1.0
871430                      Once Upon a Time  ...  1.0
871390                  Arsenic and Old Lace  ...  1.0
871425             None But the Lonely Heart  ...  1.0
871424                         Night and Day  ...  1.0
871427                             Notorious  ...  1.0
871444      The Bachelor and the Bobby-Soxer  ...  1.0
871445                     The Bishop's Wife  ...  1.0
871421  Mr. Blandings Builds His Dream House  ...  1.0
871402          Every Girl Should Be Married  ...  1.0
871411                I Was a Male War Bride  ...  1.0

[18 rows x 6 columns]

### How many roles were available for actors in the 1950s?

In [95]:
print("The number of roles available for actors in 1950s are:")
print(len(cast[(cast['year']>=1950)&(cast['year']<=1959)&(cast['type']=='actor')].type))

The number of roles available for actors in 1950s are:
157738


### How many roles were available for actresses in the 1950s?

In [96]:
print("The number of roles available for actresses in 1950s are:")
print(len(cast[(cast['year']>=1950)&(cast['year']<=1959)&(cast['type']=='actress')].type))

The number of roles available for actresses in 1950s are:
57733


### How many leading roles (n=1) were available from the beginning of film history through 1980?

In [97]:

print("The number of leading roles available from the beginning of film history through 1980 are:")
print(len(cast[(cast['year']<=1980)&(cast['n']==1)].type))

The number of leading roles available from the beginning of film history through 1980 are:
65140


### How many non-leading roles were available through from the beginning of film history through 1980?

In [98]:
print("The number of non-leading roles available from the beginning of film history through 1980 are:")
print(len(cast[(cast['year']<=1980)&(cast['n']!=1)].type))

The number of non-leading roles available from the beginning of film history through 1980 are:
1117667


### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [99]:
print("The number of roles through 1980 that were minor enough and did not warrant a numeric n rank are:")
print(len(cast[(cast['year']==1980)&(cast['n'].isnull()==True)]))

The number of roles through 1980 that were minor enough and did not warrant a numeric n rank are:
4466
